src/safeposix/syscalls/fs_calls.rs

```diff
@@ -221,32 +223,37 @@ impl Cage {
                     uid: DEFAULT_UID,
                     gid: DEFAULT_GID,
                     mode: effective_mode,
-                    linkcount: 3,
-                    refcount: 0, //2 because ., and .., as well as reference in parent directory
+                    linkcount: 3, //3 represents the directory name, itself, and reference to the parent directory
+                    refcount: 0, //because no file descriptors are pointing to it currently
                     atime: time,
                     ctime: time,
                     mtime: time,
                     filename_to_inode_dict: init_filename_to_inode_dict(newinodenum, pardirinode),
                 });
 
-                if let Inode::Dir(ref mut parentdir) =
-                    *(metadata.inodetable.get_mut(&pardirinode).unwrap())
+                //insert a reference to the file in the parent directory and update the inode attributes
+                if let Inode::Dir(ref mut parentdir) = *(metadata.inodetable.get_mut(&pardirinode).unwrap())
                 {
                     parentdir
                         .filename_to_inode_dict
                         .insert(filename, newinodenum);
                     parentdir.linkcount += 1;
+                    parentdir.ctime = time;
+                    parentdir.mtime = time;
+                    // Here, update the ctime and mtime for the parent directory as well
```

Comment:
Man pages for this call contain the update to the timestamps as well - https://docs.oracle.com/cd/E86824_01/html/E54765/mkdir-2.html